cdap-ui/app/hydrator/controllers/detail/canvas-ctrl.js

```diff
@@ -68,26 +68,30 @@ angular.module(PKG.name + '.feature.hydrator')
               rDisabled: function() {
                 return true;
               },
-              rNodeMetricsContext: function() {
-                var appParams = HydratorPlusPlusDetailRunsStore.getParams();
-                var latestRun = HydratorPlusPlusDetailRunsStore.getLatestRun();
-                var logsParams = HydratorPlusPlusDetailRunsStore.getLogsParams();
-                let runs = HydratorPlusPlusDetailRunsStore.getRuns();
+              rNodeMetricsContext: ['$stateParams', 'GLOBALS', function($stateParams, GLOBALS) {
+                let pipelineDetailStoreState = window.CaskCommon.PipelineDetailStore.getState();
+                let programType = pipelineDetailStoreState.pipelineType === GLOBALS.etlDataPipeline ? 'workflow' : 'spark';
+                let programId = pipelineDetailStoreState.pipelineType === GLOBALS.etlDataPipeline ? 'DataPipelineWorkflow' : 'DataStreamsSparkStreaming';
+
                 return {
-                  runRecord: latestRun,
-                  runs,
-                  namespace: appParams.namespace,
-                  app: appParams.app,
-                  programType: HydratorPlusPlusDetailRunsStore.getMetricProgramType(),
-                  programId: logsParams.programId
+                  runRecord: pipelineDetailStoreState.currentRun,
+                  runs: pipelineDetailStoreState.runs,
+                  namespace: $stateParams.namespace,
+                  app: pipelineDetailStoreState.name,
+                  programType,
+                  programId
                 };
-              },
-              rPlugin: ['HydratorPlusPlusNodeService', 'HydratorPlusPlusDetailNonRunsStore', 'GLOBALS', function(HydratorPlusPlusNodeService, HydratorPlusPlusDetailNonRunsStore, GLOBALS) {
+              }],
+              rPlugin: ['HydratorPlusPlusNodeService', 'HydratorPlusPlusHydratorService', 'GLOBALS', function(HydratorPlusPlusNodeService, HydratorPlusPlusHydratorService, GLOBALS) {
                 let pluginId = pluginNode.name;
-                let appType = HydratorPlusPlusDetailNonRunsStore.getAppType();
-                let artifactVersion = HydratorPlusPlusDetailNonRunsStore.getArtifact().version;
-                let sourceConnections = HydratorPlusPlusDetailNonRunsStore.getSourceConnections(pluginId);
-                let sourceNodes = HydratorPlusPlusDetailNonRunsStore.getSourceNodes(pluginId);
+                let pipelineDetailStoreState = window.CaskCommon.PipelineDetailStore.getState();
+                let appType = pipelineDetailStoreState.pipelineType;
```

Comment:
was there still `pipelineType` in the store? didn't we roll that up to the artifact?